src/test/java/automationpractice/com/features/checkout/CheckoutFeatures.java

```diff
@@ -0,0 +1,115 @@
+package automationpractice.com.features.checkout;
+
+import automationpractice.com.pages.AddressCheckoutPage;
+import automationpractice.com.pages.HomePage;
+import automationpractice.com.pages.ProductDetailsPage;
+import automationpractice.com.pages.SummaryCheckoutPage;
+import automationpractice.com.pages.OrderHistoryAndDetailsPage;
+import automationpractice.com.pages.PaymentCheckoutPage;
+import automationpractice.com.pages.ShippingCheckoutPage;
+import automationpractice.com.steps.LoginSteps;
+import cucumber.api.java.en.And;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import net.thucydides.core.annotations.Steps;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CheckoutFeatures {
+
+    private HomePage homePage;
+    private ProductDetailsPage productDetailsPage;
+    private SummaryCheckoutPage summaryCheckoutPage;
+    private AddressCheckoutPage addressCheckoutPage;
+    private ShippingCheckoutPage shippingCheckoutPage;
+    private PaymentCheckoutPage paymentCheckoutPage;
+    private OrderHistoryAndDetailsPage orderHistoryAndDetailsPage;
+
+    @Steps
+    private LoginSteps loginSteps;
+
+    @Given("^logged customer is on home page$")
+    public void loggedCustomerIsOnHomePage() {
+        loginSteps.loginAndMoveIntoMyStorePage();
+    }
+
+    @Given("^user add first item into cart$")
+    public void userAddFirstItemIntoCart() {
+        loginSteps.loginAndMoveIntoMyStorePage();
+        homePage.clickOnSelectedProductImage(0);
+        productDetailsPage.clickOnAddToCartButton();
+        productDetailsPage.clickProceedToCheckoutButton();
+    }
+
+    @When("^user change quantity on summary checkout page$")
+    public void userChangeQuantityOnSummaryCheckoutPage() {
+        summaryCheckoutPage.clickPlusAndMinusButtonGivenNumberOfTimes(4,3);
+        summaryCheckoutPage.setQuantity(summaryCheckoutPage.getCurrentQuantity());
+    }
+
+    @And("^user confirms every step from summary checkout page up to payment checkout page$")
+    public void userConfirmsEveryStepFromSummaryCheckoutPageUpToPaymentCheckoutPage() {
+        summaryCheckoutPage.clickOnProceedToCheckoutButton();
+        addressCheckoutPage.clickOnProceedToCheckoutButton();
+        shippingCheckoutPage.clickAgreeCheckbox();
+        shippingCheckoutPage.clickOnProceedToCheckoutButton();
+        paymentCheckoutPage.setQuantity(paymentCheckoutPage.getCurrentQuantity());
+    }
+
+    @Then("^summary checkout page has a given number of items in summary checkout page$")
+    public void summaryCheckoutPageHasAGivenNumberOfItemsInSummaryCheckoutPage() {
+        assertThat(summaryCheckoutPage.getQuantity()).isEqualTo(paymentCheckoutPage.getQuantity());
+    }
```

Comment:
W tej klasie powinieneś się określić jaką powinieneś mieć quantity i po prostu przyrówanać to do tego co Ci zwraca paymentCheckoutPage.